packages/backend/src/services/UserAttributesService/UserAttributeUtils.ts

```diff
@@ -82,14 +96,25 @@ export const getFilteredExplore = (
                     ...table,
                     metrics: Object.fromEntries(
                         Object.entries(table.metrics).filter(
-                            ([metricName, metric]) =>
-                                !metric.tablesReferences ||
-                                metric.tablesReferences.every(
-                                    (tableReference) =>
-                                        filteredTableNames.includes(
-                                            tableReference,
-                                        ),
-                                ),
+                            ([metricName, metric]) => {
+                                const parentDimensionIsHidden =
+                                    hiddenDimensions.some(
+                                        (dimension) =>
+                                            dimension.name ===
```

Comment:
not it checks against the metric.dimensionReference

Reply:
![Screenshot from 2024-03-06 10-34-30](https://github.com/lightdash/lightdash/assets/1983672/2adb5b42-ea74-4ca3-a130-4fd48e3200b8)

Reply:
This is still going to cause us problems as I can reference the dimension A in metrics defined under dimension B.

I propose update`exploreCompiled.ts` (pseudo code):

compileDimensionReference() - now returns the referenced dimension
```
        return {
            ...
            dimension: referencedDimension,
        };
```

compileMetricSql() - aggregates all the dimension referenced and then returns 
```
let dimensionsReferences =  new Set<{id: string, table: string, name: string}>();

...

                if(compiledReference.dimension) {
                    dimensionsReferences = ([
                        ...dimensionsReferences,
                        { id: getItemId(compiledReference.dimension), table: compiledReference.dimension.table, name: compiledReference.dimension.name}
                    ]);
                }
                
...
 
                 return { sql: compiledSql, tablesReferences, dimensionsReferences };
```

Using `{id: string, table: string, name: string}` so we have all the information possible to relate fields/tables.